### PR TITLE
Include Chainlink Version in Logger Name

### PR DESCRIPTION
**Pull request by essamhassan**

### CODE REVIEW

**Comment by jmank88** [comment]
I think we'll have to wrangle the load flags for CI too, to match the makefile:
https://github.com/smartcontractkit/chainlink/blob/develop/GNUmakefile#L47-L48
https://github.com/smartcontractkit/chainlink/blob/develop/GNUmakefile#L7-L11
https://github.com/smartcontractkit/chainlink/blob/develop/tools/bin/ldflags
CI calls this:
https://github.com/smartcontractkit/chainlink/blob/develop/tools/bin/go_core_tests
CI may only need the sha, if we want to simplify. Then `if static.Version == "unset"`, fallback to the sha. (And maybe if `CHAINLINK_DEV`, then include them both for full precision?)

**Comment by essamhassan** [comment]
I'm pro we keep it on track with production, go test seems to support -ldflags correctly so I updated the PR with the correct flags

**Comment by jmank88** [comment]
> I'm pro we keep it on par with production, `go test` seems to support `-ldflags` properly so I updated the PR with the correct flags

That's fine if it's simpler, but then we need a different way to detect when to include the hash as well, so these don't look like release logs from the last version.

**Comment by jmank88** [comment]
Or instead of worrying about the mess, we could just always include the shorter 7 character version of the hash.
`l.Named(static.Version).Named(static.Sha[7:])`

**Comment by essamhassan** [comment]
I think the latter make sense. Updated the PR

File: core/logger/logger.go [hunk 1]
```diff
@@ -177,7 +179,7 @@ func NewLogger() Logger {
 	for _, msg := range parseErrs {
 		l.Error(msg)
 	}
-	return l
+	return l.Named(fmt.Sprintf("%s@%s", static.Version, static.Sha[:7]))
```
Comment (jmank88) [inline core/logger/logger.go:182]: I wonder if we should make this a helper func? Maybe `logger.VersionName() or something?`

Comment (MStreet3) [reply core/logger/logger.go:182]: +1

File: tools/bin/go_core_tests [hunk 2]
```diff
@@ -4,7 +4,8 @@ set +e
 
 echo "Failed tests and panics: ---------------------"
 echo ""
-go test -v -p 3 -parallel 4 -coverprofile=coverage.txt -covermode=atomic ./... | tee ./output.txt | grep --line-buffered --line-number -e "\-\-\- FAIL" -e "FAIL\s"
+GO_LDFLAGS=$(bash tools/bin/ldflags)
```
[the remark this answers is not in the document]
Comment (essamhassan) [reply tools/bin/go_core_tests:7]: To pass the version/sha properly to the test binary (the same way we pass it in prod)

File: core/logger/logger.go (hunk 1 above)
Comment (essamhassan) [reply core/logger/logger.go]: I extracted only the name formatting to a pkg helper func baseLoggerName

File: tools/bin/go_core_tests (hunk 2 above)
[the remark this answers is not in the document]
Comment (MStreet3) [reply tools/bin/go_core_tests:7]: ty

File: core/logger/test_logger_test.go [hunk 3]
```diff
@@ -35,8 +37,8 @@ func TestTestLogger(t *testing.T) {
 		testMessage = "Test message"
 	)
 	lgr.Warn(testMessage)
-	// [WARN]  Test message		logger/test_logger_test.go:23    logger=TestLogger
-	requireContains("[WARN]", testMessage, fmt.Sprintf("logger=%s", testName))
+	// [WARN]  Test message		logger/test_logger_test.go:23    logger=1.0.0@sHaValue.TestLogger
+	requireContains("[WARN]", testMessage, fmt.Sprintf("logger=%s@%s.%s", static.Version, static.Sha[:7], testName))
```
Comment (jmank88) [inline core/logger/test_logger_test.go:41]: Maybe should use `baseLoggerName()` for these too? Dedupes and limits the potential slice panic to one location where it's obvious what is wrong :shrug:

Comment (essamhassan) [reply core/logger/test_logger_test.go]: makes sense, updated.

**Review by jmank88 — APPROVED** [review]
LGTM!

**Review by samsondav — APPROVED** [review]

**Pull request merged**